advanced-javascript/tests/arrays.test.js

```diff
@@ -18,43 +18,86 @@ describe('Arrays', () => {
   describe('`each`', () => {
     it('should be a function', () => {
       const each = arrayFunctions.each;
-      expect(each).to.be.a('string');
+      expect(each).to.be.a('function');
     });
     // begin here
+    it('should invoke callback of the length of array', () => {
+      const each = arrayFunctions.each;
+      const callBack = sinon.spy();
+      each([1,2,3], callBack);
+      expect(callBack).to.have.callCount(3);
+    });
   });
 
   describe('`map`', () => {
     it('should be a function', () => {
       const map = arrayFunctions.map;
-      expect(map).to.be.an('object');
+      expect(map).to.be.an('function');
+    });
+    it('should invoke callback of the length of array', () => {
+      const map = arrayFunctions.map;
+      const callBack = sinon.spy();
+      map([1,2,3], callBack);
+      expect(callBack).to.have.callCount(3);
     });
   });
 
   describe('`reduce`', () => {
     it('should be a function', () => {
       const reduce = arrayFunctions.reduce;
-      expect(reduce).to.be.a('number');
+      expect(reduce).to.be.a('function');
+    });
+    it('should invoke callback of the length of array', () => {
+      const reduce = arrayFunctions.reduce;
+      const callBack = sinon.spy();
+      const arr = [1,2,3];
+      reduce(arr, callBack, memo=0);
+      expect(callBack).to.have.callCount(3);
```

Comment:
Would like to see `reduce()` specific test assertions here (e.g., `reduce()` is returning one value from an array of numbers, `reduce()` accepts a memo argument, or `reduce()` can handle any data type)